README.md

```diff
@@ -20,153 +20,82 @@ This contract is intended to operate as the distributor, and has functionality t
-- Ensure that there is 0 token balance in the contract
 - Transfer the exact amount required for the entire program (all streams, all amounts, all periods)
-- Use `setValidatedRecipientList(streamerAddresses, amountsPerPeriod, maxPeriods)`
-- Do not load the next schedule until the current one is complete.
+- Use `addRecipients(streamerAddresses, amountsPerPeriod, maxPeriods, doNotStartBeforeTimestamp)`
 
 
 #### To abort a schedule midway through or reset
-- Use `setRecipientList([], [], [])` to clear the list.
+- Use `removeRecipients(recipients)` to one or multiple configurations from the list.
 - Use `sweep(token)` to transfer any remaining tokens back to the owner.
 - Now you can use the normal process to set a new schedule.
 
 ## Deployment and operations
 
 ### Dependancies/environment setup
-This repo requires eth-brownie to work.  Both versions 1.17 and 1.19 have been tested.  The contracts were developed using python3.9 and brownie 1.19.
-
-On a mac with homebrew from the root of this repo run the following commands to install brownie and run the tests:
-```bash
-export WEB3_INFURA_PROJECT_ID=<your infura key>
-export ETHERSCAN_TOKEN=<your etherscan API token> 
-brew install python@3.9
-python3.9 -m venv venv
-source venv/bin/activate
-pip3 install -r requirements.txt
-brownie test
-```
-Note that ETHERSCAN_TOKEN can be changed to whatever network you are operating on, for example:
+This repo requires hardhat to work.
 
- - POLYGONSCAN_TOKEN
- - OPTIMISMSCAN_TOKEN
- - ABRISCAN_TOKEN
- - ZKEVMSCAN_TOKEN
- - GNOSISSCAN_TOKEN
+#### Install
+```
+npm install
+```
 
-Note that this installs a version of brownie 1.19.3 modified/maintained by tritium to include the extra chains Balancer operates on. 
+#### Add .env
+As the injector is integrated and works with existing gauges, the local hardhat environment forks the current Polygon network.
+To do that it requires an alchemy key.
 
-You will need to have an account setup in brownie with gas.  You can read more about brownie accounts [here](https://eth-brownie.readthedocs.io/en/v1.19.0/core-accounts.html).
+```
+ALCHEMY_KEY=XXX
+```
 
-You can create a new address with
-`brownie accounts new <id>` where id is the string you will use to identify the account.
-You can view the account using `brownie accounts list`.  You will need to transfer some gas tokens to this address on the account you want to deploy on.
+#### Run tests
+```
+hardhat test
+```
 
-Read more about brownie accounts in the docs to learn how to import/use an account you already have a private key for.
+#### Adding new networks
+You can add new networks in hardhat.config.js under the network tab.
 
 ### Deploying an injector
-Deployment is done using the deploy script found in [scripts/deploy.py](scripts/deploy.py).  You will need to edit it and change the following
+Deployment is done using the deploy script found in [scripts/deploy.js](./scripts/deploy.js).  You will need to edit it and change the following
```

Comment:
The goal here was to use the factory to deploy the injectors.  So really the factory should only deployed once per chain.   This enables much easier tracking of all the injectors that exist, and for partners to launch them on their own.

Reply:
Added also a script to deploy the factory and to deploy an injector with the factory.